### PR TITLE
Improve core layer optional params management

### LINKED ISSUE #105: Simplify check in Core layer impl
followup to https://github.com/opendatahub-io/model-registry/pull/85#discussion_r1372857213

### LINKED ISSUE #110: Make use of glog in core layer
**Is your feature request related to a problem? Please describe.**
As suggested [here](https://github.com/opendatahub-io/model-registry/pull/85#discussion_r1374915916) make use of [`glog`](https://github.com/golang/glog).

**Describe the solution you'd like**
Substitute standard `log` module with `glog` one.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by lampajr**
Fixes https://github.com/opendatahub-io/model-registry/issues/105 
Fixes https://github.com/opendatahub-io/model-registry/issues/110
Furthermore it provides some optional paramas management 

## Description

Additional improvements:
- For `Update` operations, user does not have to provide the `parentResourceId` - the implementation is going to retrieve it from the original parent association.
  - Created two internal utlities: `getModelVersionByArtifactId` and `getRegisteredModelByVersionId`
- Manage all possible failures that can happen for `create/update`, based on provided input params.

## How Has This Been Tested?

Added additional test cases in `core_test` that cover the new edge cases, mostly considering possible failures.
```bash
make test-cover
```
## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: internal/core/core.go [hunk 1]
```diff
@@ -79,10 +80,26 @@ func NewModelRegistryService(cc grpc.ClientConnInterface) (ModelRegistryApi, err
...
+		}
+	}
+
+	// if already existing assure the name is the same
+	if existing != nil {
+		if registeredModel.Name == nil {
+			// user did not provide it
+			// need to set it to avoid mlmd error "context name should not be empty"
+			registeredModel.Name = existing.Name
+		}
```
Comment (tarilabs) [inline internal/core/core.go]:
```suggestion
	if existing != nil && registeredModel.Name == nil {
		// user did not provide it
		// need to set it to avoid mlmd error "context name should not be empty"
		registeredModel.Name = existing.Name
	}
```

how about? 🤔

File: internal/core/core.go [hunk 2]
```diff
@@ -200,26 +254,48 @@ func (serv *modelRegistryService) GetRegisteredModels(listOptions ListOptions) (
...
-	modelCtx, err := serv.mapper.MapFromModelVersion(modelVersion, *registeredModelIdCtxID, registeredModelName)
+
+	// if already existing assure the name is the same
+	if existing != nil {
+		if modelVersion.Name == nil {
+			// user did not provide it
+			// need to set it to avoid mlmd error "context name should not be empty"
+			modelVersion.Name = existing.Name
+		}
+	}
```
Comment (tarilabs) [inline internal/core/core.go]: Same here, how about

```suggestion
	if existing != nil && registeredModel.Name == nil {
		// user did not provide it
		// need to set it to avoid mlmd error "context name should not be empty"
		registeredModel.Name = existing.Name
	}
```

**Review by tarilabs — DISMISSED** [review]
Thank you!
Minor comments, up to you.

Later when wiring the REST, we could think of specializing error with types beyond the string-msg, so to return proper HTTP xxx error codes; wdyt?

File: internal/core/core.go (hunk 1 above)
Comment (lampajr) [reply internal/core/core.go]: good catch!! forgot to simplify it :man_facepalming:

**Comment by lampajr** [comment]
> Later when wiring the REST, we could think of specializing error with types beyond the string-msg, so to return proper HTTP xxx error codes; wdyt?

Definitely agree on that, we can create our own `Error` instances wrapping additional meaningful fields

File: internal/core/core.go (hunk 2 above)
Comment (lampajr) [reply internal/core/core.go]: hehe it should have been `modelVersion` and not `registeredModel`, fixed with 5c10d074a9537a74fc051fc5cee21ef098fcf4da

**Comment by lampajr** [comment]
There was an issue in the applied suggestion, just a typo. 
Fixed with 5c10d074a9537a74fc051fc5cee21ef098fcf4da

**Review by tarilabs — APPROVED** [review]
